x/stakeibc/keeper/hooks.go

```diff
@@ -236,11 +237,18 @@ func (k Keeper) GetUndelegatedBalance(chainId string, depositRecords []recordsty
 
 // Returns the total delegated balance that's stored in LSM tokens
 // These are identified by any status besides "DEPOSIT_PENDING"
-func (k Keeper) GetTotalTokenizedDelegations(ctx sdk.Context, chainId string) sdk.Dec {
+func (k Keeper) GetTotalTokenizedDelegations(ctx sdk.Context, hostZone types.HostZone) sdk.Dec {
 	total := sdkmath.ZeroInt()
-	for _, deposit := range k.RecordsKeeper.GetLSMDepositsForHostZone(ctx, chainId) {
+	for _, deposit := range k.RecordsKeeper.GetLSMDepositsForHostZone(ctx, hostZone.ChainId) {
 		if deposit.Status != recordstypes.LSMTokenDeposit_DEPOSIT_PENDING {
-			total = total.Add(deposit.Amount)
+			validator, _, found := GetValidatorFromAddress(hostZone.Validators, deposit.ValidatorAddress)
+			if !found {
+				k.Logger(ctx).Error(fmt.Sprintf("Validator %s found in LSMTokenDeposit but no longer exists", deposit.ValidatorAddress))
+				continue
+			}
+			liquidStakedShares := deposit.Amount
+			liquidStakedTokens := sdk.NewDecFromInt(liquidStakedShares).Mul(validator.InternalSharesToTokensRate)
+			total = total.Add(liquidStakedTokens.TruncateInt())
```

Comment:
nit: do we get better precision if we add to the total in `Dec` space, then truncate it outside the loop after all have been added? not sure

Reply:
hmm, good question. I think we do get more precision, but it might be less "accurate" since we want to replicate the number we're using when giving out stTokens